main.go

```diff
@@ -436,21 +426,198 @@ func main() {
+
+	complianceServerCtx := startComplianceEventsAPI(
+		controllerCtx,
+		cfg,
+		client,
+		complianceEventsNamespace,
+		net.JoinHostPort(eventHistoryAPIHost, eventHistoryAPIPort),
+		&wg,
+		tempDir,
+		replicatedPolicyUpdates,
+	)
+
+	replicatedPolicyCtrler := &propagatorctrl.ReplicatedPolicyReconciler{
+		Propagator:          propagator,
+		ResourceVersions:    replicatedResourceVersions,
+		DynamicWatcher:      dynamicWatcher,
+		TemplateResolver:    templateResolver,
+		ComplianceServerCtx: complianceServerCtx,
+	}
+
+	if err = (replicatedPolicyCtrler).SetupWithManager(
+		mgr, replPolicyMaxConcurrency, dynamicWatcherSource, replicatedUpdatesSource, templatesSource,
+	); err != nil {
+		log.Error(err, "Unable to create the controller", "controller", "replicated-policy")
+		os.Exit(1)
+	}
+
 	log.Info("Starting manager")
 
-	historyAPIAddr := net.JoinHostPort(eventHistoryAPIHost, eventHistoryAPIPort)
+	wg.Add(1)
+
+	go func() {
+		if err := mgr.Start(controllerCtx); err != nil {
+			log.Error(err, "Problem running manager")
+			os.Exit(1)
+		}
+
+		wg.Done()
+	}()
 
-	if enableComplianceEventsStore {
-		err = complianceeventsapi.StartManager(controllerCtx, cfg, enableLeaderElection, historyAPIAddr)
+	wg.Wait()
+}
+
+func startComplianceEventsAPI(
+	ctx context.Context,
+	cfg *rest.Config,
+	client *kubernetes.Clientset,
+	controllerNamespace string,
+	complianceAPIAddr string,
+	wg *sync.WaitGroup,
+	tempDir string,
+	reconcileRequests chan<- event.GenericEvent,
+) *complianceeventsapi.ComplianceServerCtx {
+	var dbConnectionURL string
+
+	dbSecret, err := client.CoreV1().Secrets(controllerNamespace).Get(
+		ctx, complianceeventsapi.DBSecretName, metav1.GetOptions{},
+	)
+	if k8serrors.IsNotFound(err) {
+		log.Info(
+			"Could not start the compliance events API. To enable this functionality, ensure the Postgres "+
+				"connection secret is valid in the controller namespace.",
+			"secretName", complianceeventsapi.DBSecretName,
+			"namespace", controllerNamespace,
+		)
+	} else if err != nil {
+		log.Error(
+			err,
+			"Failed to determine if the secret was defined",
+			"secretName", complianceeventsapi.DBSecretName,
+			"namespace", controllerNamespace,
+		)
+
+		os.Exit(1)
+	} else {
+		var err error
+
+		dbConnectionURL, err = complianceeventsapi.ParseDBSecret(dbSecret, tempDir)
+		if err != nil {
+			log.Error(
+				err,
+				"Fix the connection details to enable the compliance events API feature",
+				"secret", complianceeventsapi.DBSecretName,
+				"namespace", controllerNamespace,
+			)
+		}
+	}
+
+	complianceServerCtx, err := complianceeventsapi.NewComplianceServerCtx(dbConnectionURL)
+	if err == nil {
+		// If the migration failed, MigrateDB will log it and MonitorDatabaseConnection will fix it.
+		err := complianceServerCtx.MigrateDB(ctx, client, controllerNamespace)
 		if err != nil {
-			log.Error(err, "Ignoring since this is technical preview")
+			log.Info("Will periodically retry the migration until it is successful")
```

Comment:
Why it is no` log.Error(err`

Reply:
In a call, Matt mentioned that if this used `log.Error`, it would give a stack trace kind of thing every time. If the DB is down, this is expected to be called a lot, so then the logs would be extremely noisy with less useful information. Using `log.Info` keeps it shorter.